### PR TITLE
Use condense_control()

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
This PR reflects the changes made in https://github.com/tidymodels/parsnip/pull/808.

This PR should close https://github.com/tidymodels/tune/issues/449, the error isn't perfect, but it is a lot better

``` r
library(tidymodels)

rec_no_tune_1 <-
  recipe(mpg ~ ., data = mtcars) %>%
  step_normalize(all_predictors())

set.seed(363)
mt_folds <- vfold_cv(mtcars, v = 5)

glmn_mod <- decision_tree(tree_depth = tune(), min_n = 2) %>% 
  set_mode("regression") %>%
  set_engine("rpart")

workflow() %>%
  add_recipe(rec_no_tune_1) %>%
  add_model(glmn_mod) %>%
  tune_bayes(mt_folds, control = control_grid())
#> Error in `parsnip::condense_control()` at tune/R/tune_bayes.R:195:4:
#> ! Object of class `control_grid` cannot be corresed to object of class `control_bayes`.
#> • The following arguments are missing:
#> • 'no_improve', 'uncertain', 'seed', 'time_limit', and 'save_gp_scoring'
```

<sup>Created on 2022-09-14 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

This means that we get more flexibility in cases where we need to pass a control object from function to function.

``` r
library(tidymodels)

rec_no_tune_1 <-
  recipe(mpg ~ ., data = mtcars) %>%
  step_normalize(all_predictors())

mt_folds <- vfold_cv(mtcars, v = 5)

tree_mod_no_tune <- decision_tree(min_n = 3) %>% 
  set_mode("regression") %>%
  set_engine("rpart")

tree_wf_no_tune <- workflow() %>%
  add_recipe(rec_no_tune_1) %>%
  add_model(tree_mod_no_tune)

# fit_resamples
tree_wf_no_tune %>%
  fit_resamples(mt_folds, control = control_bayes())
#> # Resampling results
#> # 5-fold cross-validation 
#> # A tibble: 5 × 4
#>   splits         id    .metrics         .notes          
#>   <list>         <chr> <list>           <list>          
#> 1 <split [25/7]> Fold1 <tibble [2 × 4]> <tibble [0 × 3]>
#> 2 <split [25/7]> Fold2 <tibble [2 × 4]> <tibble [0 × 3]>
#> 3 <split [26/6]> Fold3 <tibble [2 × 4]> <tibble [0 × 3]>
#> 4 <split [26/6]> Fold4 <tibble [2 × 4]> <tibble [0 × 3]>
#> 5 <split [26/6]> Fold5 <tibble [2 × 4]> <tibble [0 × 3]>
```

<sup>Created on 2022-09-14 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

### CODE REVIEW

**Comment by EmilHvitfeldt** [comment]
As far as I can tell, the CI errors are fixed in https://github.com/tidymodels/tune/pull/541

File: tests/testthat/test-condense_control.R [hunk 1]
```diff
@@ -0,0 +1,52 @@
+test_that("control functions respect hierarchy with condense_control", {
+
+  ctrl <- parsnip::condense_control(
```
Comment (EmilHvitfeldt) [inline tests/testthat/test-condense_control.R:3]: I'm assume here that `control_grid()` is in theory a super of `control_resamples()`, they just happen to be equal at this point in time

File: NEWS.md [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # tune (development version)
 
+* `last_fit()`, `fit_resamples()`, `tune_grid()`, and `tune_bayes()` no longer error if `control` argument isn't a the corresponding `control_*()` object. Will work as long as the object passed to `control` includes the same elements as the required `control_*()` object.
```
Comment (topepo) [inline NEWS.md:3]:
```suggestion
* `last_fit()`, `fit_resamples()`, `tune_grid()`, and `tune_bayes()` do not automatically error if  the wrong type of `control` object is passed. If the passed control object is not a superset of the one that is needed, the function will still error. As an example, passing `control_grid()` to `tune_bayes()` will fail but passing `control_bayes()` to `tune_grid()` will not. ([#449](https://github.com/tidymodels/tune/issues/449))
```

**Review by topepo — CHANGES_REQUESTED** [review]

**Comment by EmilHvitfeldt** [comment]
Changes has been merged :)

**Pull request merged**